### PR TITLE
Reapply the submodule updates to LLVM, wasi-libc and config. And fixes the build

### PR DESCRIPTION
**Pull request by lygstate**
It appears I accidentally merged in a git submodule diff in #197
which reverted the LLVM, wasi-libc, and config updates. Revert
that change, updating wasi-sdk again to LLVM 13.0 and the latest
wasi-libc and config.

Also fixes the build on windows after upgrade the sub-projects

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -154,8 +155,9 @@ build/libcxx.BUILT: build/llvm.BUILT build/compiler-rt.BUILT build/wasi-libc.BUI
 	# Do the build.
 	mkdir -p build/libcxx
 	cd build/libcxx && cmake -G Ninja $(LIBCXX_CMAKE_FLAGS) \
-	    -DCMAKE_C_FLAGS="$(DEBUG_PREFIX_MAP) --sysroot=$(BUILD_PREFIX)/share/wasi-sysroot" \
-	    -DCMAKE_CXX_FLAGS="$(DEBUG_PREFIX_MAP) --sysroot=$(BUILD_PREFIX)/share/wasi-sysroot" \
+	    -DCMAKE_SYSROOT=$(BUILD_PREFIX)/share/wasi-sysroot \
+	    -DCMAKE_C_FLAGS="$(DEBUG_PREFIX_MAP) \
+	    -DCMAKE_CXX_FLAGS="$(DEBUG_PREFIX_MAP) \
```
Comment (sbc100) [inline Makefile:160]: Why these changes?

Comment (lygstate) [reply Makefile:160]: This is the reliable way to pass sysroot to the compiler by cmake.
It's can  handling msys2 and msys2/mingw properly

Comment (lygstate) [reply Makefile]: I split it out into other PR

**Comment by sbc100** [comment]
lgtm.   Can you give the PR a better title and description?

**Comment by sunfishcode** [comment]
Looks good, thanks for your help!
